cmake/FindFFnvcodec.cmake

```diff
@@ -0,0 +1,98 @@
+#[=======================================================================[.rst
+FindFFnvcodec
+-------------
+
+FindModule for FFnvcodec and the associated library
+
+Imported Targets
+^^^^^^^^^^^^^^^^
+
+This module defines the :prop_tgt:`IMPORTED` target ``FFnvcodec::FFnvcodec``.
+
+Result Variables
+^^^^^^^^^^^^^^^^
+
+This module sets the following variables:
+
+``FFnvcodec_FOUND``
+  True, if the library was found.
+``FFnvcodec_VERSION``
+  Detected version of found FFnvcodec library.
+
+Cache variables
+^^^^^^^^^^^^^^^
+
+The following cache variables may also be set:
+
+``FFnvcodec_INCLUDE_DIR``
+  Directory containing ``nvEncodeAPI.h``.
+
+Distributed under the MIT License, see accompanying LICENSE file or
+https://github.com/PatTheMav/cmake-finders/blob/master/LICENSE for details.
+(c) 2023 Trystan Mata
+
+#]=======================================================================]
+
+# cmake-format: off
+# cmake-lint: disable=C0103
+# cmake-lint: disable=C0301
+# cmake-format: on
+
+include(FindPackageHandleStandardArgs)
+
+find_package(PkgConfig QUIET)
+if(PKG_CONFIG_FOUND)
+  pkg_search_module(PC_FFnvcodec QUIET ffnvcodec)
+endif()
+
+find_path(
+  FFnvcodec_INCLUDE_DIR
+  NAMES ffnvcodec/nvEncodeAPI.h
+  HINTS ${PC_FFnvcodec_INCLUDE_DIRS}
+  PATHS /usr/include /usr/local/include
+  DOC "FFnvcodec include directory")
+
+if(PC_FFnvcodec_VERSION VERSION_GREATER 0)
+  set(FFnvcodec_VERSION ${PC_FFnvcodec_VERSION})
+elseif(EXISTS "${FFnvcodec_INCLUDE_DIR}/ffnvcodec/nvEncodeAPI.h")
+  file(STRINGS "${FFnvcodec_INCLUDE_DIR}/ffnvcodec/nvEncodeAPI.h" _version_string
+       REGEX "^.*NVENCAPI_(MAJOR|MINOR)_VERSION[ \t]+[0-9]+[ \t]*$")
+
+  string(REGEX REPLACE ".*MAJOR_VERSION[ \t]+([0-9]+).*" "\\1" _version_major "${_version_string}")
+  string(REGEX REPLACE ".*MINOR_VERSION[ \t]+([0-9]+).*" "\\1" _version_minor "${_version_string}")
+
+  set(FFnvcodec_VERSION "${_version_major}.${_version_minor}")
+  unset(_version_major)
+  unset(_version_minor)
+else()
+  if(NOT FFnvcodec_FIND_QUIETLY)
+    message(AUTHOR_WARNING "Failed to find FFnvcodec version.")
+  endif()
+  set(FFnvcodec_VERSION 0.0.0)
+endif()
+
+if(CMAKE_HOST_SYSTEM_NAME MATCHES "Darwin|Windows")
+  set(FFnvcodec_ERROR_REASON "Ensure that a FFnvcodec distribution is provided as part of CMAKE_PREFIX_PATH.")
+elseif(CMAKE_HOST_SYSTEM_NAME MATCHES "Linux|FreeBSD")
+  set(FFnvcodec_ERROR_REASON "Ensure FFnvcodec library is available in local include paths.")
```

Comment:
See above.

Reply:
Also include paths do not provide a library (that'd be library search paths).

If the headers are part of a binary distribution, then installation of the library itself should make them available.

Reply:
Changed wording to be similar to AMF find module.